packages/salesforce-adapter/src/adapter.ts

```diff
@@ -347,6 +348,10 @@ export default class SalesforceAdapter implements AdapterOperations {
       .concat(makeArray(config.instancesRegexSkippedList))
       .map(e => new RegExp(e))
     this.maxItemsInRetrieveRequest = config.maxItemsInRetrieveRequest ?? maxItemsInRetrieveRequest
+
+    this.deletionPolicy = _.isUndefined(config.client?.deploy?.postDelete)
+      || config.client?.deploy?.postDelete ? 'destructiveChangesPost' : 'destructiveChanges'
```

Comment:
reversing the condition would make it a bit simpler
```suggestion
    this.deletionPolicy = config.client?.deploy?.postDelete === false ? 'destructiveChanges' : 'destructiveChangesPost'
```
though probably this is not the correct condition for the layer of code anyway because these names are not supposed to be at this level

Reply:
I think it might be more confusing this way (more likely to be replaced with `!config.client?.deploy?.postDelete`)